customize_image.sh

```diff
@@ -314,27 +312,19 @@ if [ -e "$IMAGEDIR/etc/init.d/apache2" ] ; then
 fi
 
 if [ "$MINIMG" != "1" ]; then
-	# fake a revpi procfs ... otherwise NodeRed installer will get confused
-	for procfile in meminfo cpuinfo; do
-		touch "$IMAGEDIR/proc/$procfile"
-		mount -o ro,bind "$BAKERYDIR/templates/fake-procfs/$procfile" "$IMAGEDIR/proc/$procfile"
-	done
-
-	# install nodejs and nodered with an install script and revpi-nodes from npm repository
-	NODEREDSCRIPT="/tmp/update-nodejs-and-nodered.sh"
-	/usr/bin/curl -sL \
-		https://raw.githubusercontent.com/node-red/linux-installers/master/deb/update-nodejs-and-nodered\
-		--output "$IMAGEDIR/$NODEREDSCRIPT"
-	chmod 755 "$IMAGEDIR/$NODEREDSCRIPT"
-	NODERED_VER="3.0.2"
-	chroot "$IMAGEDIR" /usr/bin/sudo -u pi $NODEREDSCRIPT --confirm-install --confirm-pi --no-init --nodered-version="$NODERED_VER" --node18
-	rm "$IMAGEDIR/$NODEREDSCRIPT"
-
-	# remove fake procfs after NodeRed setup
-	for procfile in meminfo cpuinfo; do
-		umount "$IMAGEDIR/proc/$procfile"
-		rm "$IMAGEDIR/proc/$procfile"
-	done
+	# Use NodeJS LTS 18 sources
+	cp "$BAKERYDIR/templates/nodered/nodesource.gpg" "$IMAGEDIR/etc/apt/trusted.gpg.d"
+	cp "$BAKERYDIR/templates/nodered/nodesource.list" "$IMAGEDIR/etc/apt/sources.list.d"
+
+	# Install nodejs 18 including npm
+	chroot "$IMAGEDIR" apt-get update
+	chroot "$IMAGEDIR" apt-get -y install nodejs
+
+	# Install node-red via npm as explained in the node-red documentation
+	chroot "$IMAGEDIR" npm install -g --unsafe-perm node-red || true
```

Comment:
This is quite fragile without a version number specified and will make it impossible to avoid a major version bump during a release family. That was the reason, why we decided to pin the nodered version with a specific version.

Reply:
I restored the old behavior by using `NODERED_VER`.